src/kem/kyber/compress.rs

```diff
@@ -1,48 +1,21 @@
-use super::{
-    arithmetic::{KyberFieldElement, KyberPolynomialRingElement},
-    constants::FIELD_MODULUS,
-    conversions::to_unsigned_representative,
-};
+use super::{arithmetic::KyberFieldElement, constants::FIELD_MODULUS};
 
-pub(super) fn compress_q<const COEFFICIENT_BITS: usize>(fe: u16) -> KyberFieldElement {
-    debug_assert!(COEFFICIENT_BITS <= 11);
-    debug_assert!(i32::from(fe) < FIELD_MODULUS);
-
-    let mut compressed = (fe as u32) << (COEFFICIENT_BITS + 1);
+#[cfg_attr(hax, hax_lib_macros::requires(coefficient_bits <= 11 && i32::from(fe) <= FIELD_MODULUS))]
+#[cfg_attr(hax, hax_lib_macros::ensures(|result| result >= 0 && result <= (1 << coefficient_bits) - 1))]
+pub(super) fn compress_q(coefficient_bits: usize, fe: u16) -> KyberFieldElement {
```

Comment:
What's the reason to use arguments instead of const generics for these?

Reply:
hax does not like a const-generic right here. it will likely be fixed "soon-ish" but this change allows this module to be typechecked in F*.